### ISSUE TITLE
Typical irritating W10 experience with installation

### ISSUE THREAD

**Issue by Mrodent** [pwsacademy/swift-setup#8]
Hi,
I have to use W10 for a Swift application. You're probably not particularly interested in helping W10 users install Swift but sometimes we have no choice regarding OS.

Your W10 installation page says "Next, install or update App Installer from the Microsoft Store:" I had never heard of the "Microsoft Store" and assumed it would be a typical Microsoft experience. It was: when I found the "Microsoft Store" I searched for "App Installer".  Lots of crap came up, but not the thing I was looking for. Something called "APK App Installer" came up. Obviously I'm tempted to wonder if this is what you meant me to install. But "APK" is some kind of Android dev package or something. 

I also ran "> winget" in a command prompt: "not recognised". 

Eventually I found this page: https://learn.microsoft.com/en-us/windows/msix/app-installer/install-update-app-installer It has a download button. You might want to tell anyone who is unlucky enough to have to interact with Microsoft how to install "app installer"/"winget".

- after downloading that file, which was called Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle, I had to do the following
- there was no native way to "run" or "install" that file, and that extension wasn't recognised.
- "App Installer" didn't appear to be on the system, and "> winget" at the command prompt returned "unrecognised"
- then I got a command prompt running in D:\Downloads
- I entered "> powershell"
- this responded as follows:
```
D:\Downloads>powershell
Windows PowerShell                                                                                                                              Copyright (C) Microsoft Corporation. All rights reserved.                                                                                                                                                                                                                                       Try the new cross-platform PowerShell https://aka.ms/pscore6     

```
I then had to run these commands:
```
PS D:\Downloads> Add-AppXPackage -Path .\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle # i.e. the downloaded msixbundle
PS D:\Downloads> winget
Windows Package Manager v1.8.1911
Copyright (c) Microsoft Corporation. All rights reserved.

The winget command line utility enables installing applications and other packages from the command line.

usage: winget  [<command>] [<options>]

The following commands are available:
  install    Installs the given package
  ...

```

**Comment by svanimpe** [comment]
We definitely do need to support Windows 10. I don't have access to a Windows 10 machine anymore, but it seems like Microsoft moved some things around 😕

I will try to fix the instructions with the next update.

**Comment by Mrodent** [comment]
Thanks for the quick reply! Just finishing off the hoops I had to jump through in all their gory details.

**Comment by svanimpe** [comment]
I've set up Windows 10 in a virtual machine to update the instructions.

The official installation instructions using winget (https://www.swift.org/install/windows/winget/) work for me, provided that you install winget first by updating App Installer.

Regarding App Installer:

- I was able to install the MSIX bundle you mentioned just by double-clicking it.
- On the Microsoft Store, it can be found at https://apps.microsoft.com/detail/9nblggh4nns1
- Note that this only applies to Windows 10. On Windows 11, winget is included by default.

I will push updated instructions soon, but I'm waiting for Swift 6.0.2 to fix https://github.com/swiftlang/swift/issues/77097
